src/opera/parser/tosca/csar.py

```diff
@@ -0,0 +1,101 @@
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from zipfile import ZipFile
+
+import yaml
+from opera.error import ParseError
+
+
+class CloudServiceArchive:
+    def __init__(self, csar_name, csar_folder_path):
+        self._csar_name = csar_name
+        self._csar_folder_path = csar_folder_path
+        self._tosca_meta = None
+        self._root_yaml_template = None
+        self._metadata = None
+
+    def validate_csar(self):
+        with TemporaryDirectory(dir=self._csar_folder_path) as tempdir:
+            ZipFile(self._csar_name, 'r').extractall(tempdir)
+
+            try:
+                if Path(tempdir).joinpath("TOSCA-Metadata").joinpath("TOSCA.meta").exists():
+                    meta_file = open(Path(tempdir).joinpath("TOSCA-Metadata").joinpath("TOSCA.meta"), 'r')
+                    self._tosca_meta = yaml.safe_load(meta_file)
+                    meta_file.close()
```

Comment:
```suggestion
                meta_file_path = Path(tempdir) / "TOSCA-Metadata" / "TOSCA.meta"
                if meta_file_path.exists():
                    with meta_file_path.open() as meta_file:
                        self._tosca_meta = yaml.safe_load(meta_file)
```

Reply:
Shortens the lines, makes the pathlib usage consistent accross the project, and uses a safe way to handle files.

Reply:
Good point, I've added your suggested changes.